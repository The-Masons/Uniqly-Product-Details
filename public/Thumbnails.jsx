class Thumbnails extends React.Component {
  constructor(props) {
    super(props);
  }

  render() {
    var result = this.props.images.map((image, index) => {
      return <Thumbnail handleClick={this.props.handleClick} image={image} key={index}/>
    })
    return (
      <div className="thumbnail-parent">
        <button className="thumbnail-scroll-button"></button>
          <div className="thumbnails">
            {result}
          </div>
        <button className="thumbnail-scroll-button bottom"></button>
      </div>
    );
  }
}

class Thumbnail extends React.Component {
  render() {
    return (
      <div>
        <img
        className="product-thumbnail" 
        onClick={() => this.props.handleClick(this.props.image)}
        src={this.props.image.img_url}/>
      </div>
    )
  }
}

export default Thumbnails;