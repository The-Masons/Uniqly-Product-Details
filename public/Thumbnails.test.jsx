import React from 'react';
import Thumbnails from './Thumbnails.jsx';

const images = [
  "https://vignette.wikia.nocookie.net/dragonballfanon/images/7/70/Random.png/revision/latest?cb=20161221030547",
  "https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcRHjAfupBUR-cZbyUu0Lc5alrBdwc0U6G5XqibuHoiyZddzWPU4",
  "https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcR7rGURD9A_TmE86bph6WJbZ_JRPHrsqqtVrSrVPi9OwuV2Ily30w",
  "https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcTOuCGDeH07tQxcIO4WJ2RK1kdIf8ipnZ06iBGzdmirYvKU9U5v"
];
const primaryImage = "https://vignette.wikia.nocookie.net/dragonballfanon/images/7/70/Random.png/revision/latest?cb=20161221030547";

describe('Thumbnails', () => {
  test('should render with the correct items', () => {
    const thumbnails = mount(
      <Thumbnails
        handleClick={jest.fn()} 
        images={images} 
        primaryImage={primaryImage}
        />
      );
    expect(thumbnails).toMatchSnapshot();
  });

  test('should render with the correct items', () => {
    const thumbnails = mount(
      <Thumbnails
        handleClick={jest.fn()} 
        images={images} 
        primaryImage={primaryImage}
        />
      );
    expect(thumbnails).toMatchSnapshot();
  });
});