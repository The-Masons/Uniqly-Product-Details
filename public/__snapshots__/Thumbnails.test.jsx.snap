// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Thumbnails should render with the correct items 1`] = `
<Thumbnails
  handleClick={[MockFunction]}
  images={
    Array [
      "https://vignette.wikia.nocookie.net/dragonballfanon/images/7/70/Random.png/revision/latest?cb=20161221030547",
      "https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcRHjAfupBUR-cZbyUu0Lc5alrBdwc0U6G5XqibuHoiyZddzWPU4",
      "https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcR7rGURD9A_TmE86bph6WJbZ_JRPHrsqqtVrSrVPi9OwuV2Ily30w",
      "https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcTOuCGDeH07tQxcIO4WJ2RK1kdIf8ipnZ06iBGzdmirYvKU9U5v",
    ]
  }
  primaryImage="https://vignette.wikia.nocookie.net/dragonballfanon/images/7/70/Random.png/revision/latest?cb=20161221030547"
>
  <div
    className="thumbnail-parent"
  >
    <img
      className="thumbnail-scroll-up invisible"
      id="thumbnail-scroll-up"
      src="https://i.imgur.com/Q6KLD3s.png"
    />
    <div
      className="thumbnails"
    >
      <Thumbnail
        class="product-thumbnail selected-thumbnail"
        handleClick={[MockFunction]}
        image="https://vignette.wikia.nocookie.net/dragonballfanon/images/7/70/Random.png/revision/latest?cb=20161221030547"
        key="0"
      >
        <div>
          <img
            className="product-thumbnail selected-thumbnail"
            onClick={[Function]}
          />
        </div>
      </Thumbnail>
      <Thumbnail
        class="product-thumbnail"
        handleClick={[MockFunction]}
        image="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcRHjAfupBUR-cZbyUu0Lc5alrBdwc0U6G5XqibuHoiyZddzWPU4"
        key="1"
      >
        <div>
          <img
            className="product-thumbnail"
            onClick={[Function]}
          />
        </div>
      </Thumbnail>
      <Thumbnail
        class="product-thumbnail"
        handleClick={[MockFunction]}
        image="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcR7rGURD9A_TmE86bph6WJbZ_JRPHrsqqtVrSrVPi9OwuV2Ily30w"
        key="2"
      >
        <div>
          <img
            className="product-thumbnail"
            onClick={[Function]}
          />
        </div>
      </Thumbnail>
      <Thumbnail
        class="product-thumbnail"
        handleClick={[MockFunction]}
        image="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcTOuCGDeH07tQxcIO4WJ2RK1kdIf8ipnZ06iBGzdmirYvKU9U5v"
        key="3"
      >
        <div>
          <img
            className="product-thumbnail"
            onClick={[Function]}
          />
        </div>
      </Thumbnail>
    </div>
    <img
      className="thumbnail-scroll-down invisible"
      id="thumbnail-scroll-down"
      src="https://i.imgur.com/Q6KLD3s.png"
    />
  </div>
</Thumbnails>
`;

exports[`Thumbnails should render with the correct items 2`] = `
<Thumbnails
  handleClick={[MockFunction]}
  images={
    Array [
      "https://vignette.wikia.nocookie.net/dragonballfanon/images/7/70/Random.png/revision/latest?cb=20161221030547",
      "https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcRHjAfupBUR-cZbyUu0Lc5alrBdwc0U6G5XqibuHoiyZddzWPU4",
      "https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcR7rGURD9A_TmE86bph6WJbZ_JRPHrsqqtVrSrVPi9OwuV2Ily30w",
      "https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcTOuCGDeH07tQxcIO4WJ2RK1kdIf8ipnZ06iBGzdmirYvKU9U5v",
    ]
  }
  primaryImage="https://vignette.wikia.nocookie.net/dragonballfanon/images/7/70/Random.png/revision/latest?cb=20161221030547"
>
  <div
    className="thumbnail-parent"
  >
    <img
      className="thumbnail-scroll-up invisible"
      id="thumbnail-scroll-up"
      src="https://i.imgur.com/Q6KLD3s.png"
    />
    <div
      className="thumbnails"
    >
      <Thumbnail
        class="product-thumbnail selected-thumbnail"
        handleClick={[MockFunction]}
        image="https://vignette.wikia.nocookie.net/dragonballfanon/images/7/70/Random.png/revision/latest?cb=20161221030547"
        key="0"
      >
        <div>
          <img
            className="product-thumbnail selected-thumbnail"
            onClick={[Function]}
          />
        </div>
      </Thumbnail>
      <Thumbnail
        class="product-thumbnail"
        handleClick={[MockFunction]}
        image="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcRHjAfupBUR-cZbyUu0Lc5alrBdwc0U6G5XqibuHoiyZddzWPU4"
        key="1"
      >
        <div>
          <img
            className="product-thumbnail"
            onClick={[Function]}
          />
        </div>
      </Thumbnail>
      <Thumbnail
        class="product-thumbnail"
        handleClick={[MockFunction]}
        image="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcR7rGURD9A_TmE86bph6WJbZ_JRPHrsqqtVrSrVPi9OwuV2Ily30w"
        key="2"
      >
        <div>
          <img
            className="product-thumbnail"
            onClick={[Function]}
          />
        </div>
      </Thumbnail>
      <Thumbnail
        class="product-thumbnail"
        handleClick={[MockFunction]}
        image="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcTOuCGDeH07tQxcIO4WJ2RK1kdIf8ipnZ06iBGzdmirYvKU9U5v"
        key="3"
      >
        <div>
          <img
            className="product-thumbnail"
            onClick={[Function]}
          />
        </div>
      </Thumbnail>
    </div>
    <img
      className="thumbnail-scroll-down invisible"
      id="thumbnail-scroll-down"
      src="https://i.imgur.com/Q6KLD3s.png"
    />
  </div>
</Thumbnails>
`;
